src/components/button/button.test.ts

```diff
@@ -0,0 +1,70 @@
+import path from 'path';
+import type { ElementHandle, PleasantestUtils } from 'pleasantest';
+import { withBrowser } from 'pleasantest';
+import { loadTwigTemplate } from '../../../test-utils';
+
+const buttonMarkup = loadTwigTemplate(path.join(__dirname, './button.twig'));
+// Helper to test the JS-driven toggling button
+const initTogglingButtonJS = (
+  utils: PleasantestUtils,
+  togglingButton: ElementHandle
+) =>
+  utils.runJS(
+    `
+    import { initTogglingButton } from './toggling-button'
+    export default (togglingButton) => initTogglingButton(togglingButton)
+    `,
+    [togglingButton]
+  );
+
+test(
+  'should render a button element by default',
+  withBrowser(async ({ utils, screen }) => {
+    // If I don't pass _something_ to buttonMarkup(), I get an error.
+    // To get around that, I pass an empty object.
+    await utils.injectHTML(await buttonMarkup({}));
+
+    const button = await screen.getByRole('button');
+    await expect(button).toBeVisible();
+  })
+);
+
+test(
+  'should render an anchor element when href is provided',
+  withBrowser(async ({ utils, screen }) => {
+    await utils.injectHTML(await buttonMarkup({ href: '#' }));
+
+    const linkButton = await screen.getByRole('link');
+    await expect(linkButton).toBeVisible();
+  })
+);
+
+test(
+  'should render a button when aria_pressed is set',
+  withBrowser(async ({ utils, screen }) => {
+    await utils.injectHTML(
+      await buttonMarkup({ tag_name: 'a', aria_pressed: 'false' })
```

Comment:
What does the `tag_name` do in this case?

Reply:
@calebeby My intent was to test that `aria_pressed` has a higher priority over setting the `tag_name` value in the template logic. The element should be a `button` anytime `aria_pressed` is present and valid.

Reply:
Oh, that makes sense. Can you put a comment there?

Reply:
Done! https://github.com/cloudfour/cloudfour.com-patterns/pull/1318/commits/ff473746650ae1f6d1dd8c1b720376448b8a8122